app/src/main/java/org/shadowice/flocke/andotp/Activities/MainActivity.java

```diff
@@ -368,6 +378,60 @@ public void run() {
         }
     }
 
+    private String readPasswordFromNFC() {
+        if (NfcAdapter.ACTION_NDEF_DISCOVERED.equals(getIntent().getAction())) {
+            String type = getIntent().getType();
+            if (ClipDescription.MIMETYPE_TEXT_PLAIN.equals(type)) {
+                Tag tag = getIntent().getParcelableExtra(NfcAdapter.EXTRA_TAG);
+                Ndef ndef = Ndef.get(tag);
+                if (ndef == null) {
+                    // NDEF is not supported by this Tag.
+                    return null;
+                }
+                NdefMessage ndefMessage = ndef.getCachedNdefMessage();
```

Comment:
It looks like `getCachedNdefMessage()` here could potentially return null, so we might want to check for that to be safe: https://developer.android.com/reference/android/nfc/tech/Ndef#getCachedNdefMessage()